packages/aws-amplify-react/src/Widget/TOTPSetupComp.tsx

```diff
@@ -131,10 +131,11 @@ export default class TOTPSetupComp extends Component<
 		Auth.verifyTotpToken(user, totpCode)
 			.then(() => {
 				// set it to preferred mfa
-				Auth.setPreferredMFA(user, 'TOTP');
-				this.setState({ setupMessage: 'Setup TOTP successfully!' });
-				logger.debug('set up totp success!');
-				this.triggerTOTPEvent('Setup TOTP', 'SUCCESS', user);
+				Auth.setPreferredMFA(user, 'TOTP').then(() => {
+					this.setState({ setupMessage: 'Setup TOTP successfully!' });
+					logger.debug('set up totp success!');
+					this.triggerTOTPEvent('Setup TOTP', 'SUCCESS', user);
+				});
```

Comment:
In case `Auth.setPreferredMFA` is rejected is not going to be catched. it requieres a catch here
```suggestion
				}).catch(err => this.setState({ setupMessage: 'Setup TOTP failed!' }));
```

Reply:
I dont believe this is missing correct? Cause the below code has the `.catch`

Reply:
We need to catch that separately.